README.md
cooluri
=======

GIT repository for TYPO3 extension CoolUri


ChangeLog
CoolUri 1.0.33:
Link fix module
Bug #40092: Link to the CoolUri manual on docs.google removed
Bug #25947: Mountpoint page name fix
Bug #9579: Mountpoint domain fix
Bug #6080 fix - conversion of | added
Bug #38316 fix - original params are now passed to user_func
Bug #39090 fix - convert encoded [] when redirecting from old to new
Bug #38054 fix - hook changed to prevent double content generation
Bug #44729 fix - key length
Bug #3360: New feature - defaults - default values for not set parameters...
Bug #35799 fix - $_SERVER replaced by getIndpEnv
Bug #35799 fix - $_SERVER replaced by getIndpEnv
Bug #45916 fix - random shortcut now works correctly and ignores not in menu pages

CoolUri 1.0.29:
- Languages under different domains support

CoolUri 1.0.28:
- #31732: Added speaking URL to page language overlay, added validation
- LL_db from PHP to XML - 4.6 compatibility fix
- Zero value parameters translation fixed
- #31706: Decode all special chars, not only []

CoolUri 1.0.26:
- Bug #28519: Encode and decode brackets because of 4.5.4
- 301 redirect for redirectOldLinksToNew

r48786 | bednee | 2011-06-16 22:36:33 CEST

Bug #6425 fix (felogin does not work with saltedpasswords and rsaauth if cooluri is turned on)
Bug #11080 fix (config.absRefPrefix is not taken into account)
----------------------------------------------------------------------------
r48098 | bednee | 2011-05-24 23:03:23 CEST

Untranslated parameters are not stored in cache
----------------------------------------------------------------------------
r46292 | bednee | 2011-04-06 20:12:40 CEST

Shortcut problem fix
----------------------------------------------------------------------------
r46212 | bednee | 2011-04-04 21:31:10 CEST

Fixed multidomain bug + prep for lang_cooluri ext
----------------------------------------------------------------------------
r43728 | bednee | 2011-02-16 21:02:47 CET

fix for #11347: Check for doktype when following shortcuts

----------------------------------------------------------------------------
r43726 | bednee | 2011-02-16 20:48:58 CET

fixing Bug #13070: Missing domain when redirecting old links
----------------------------------------------------------------------------
r43718 | bednee | 2011-02-16 19:40:57 CET

fixing Bug #13083: Update action in backend is broken
----------------------------------------------------------------------------
r43590 | bednee | 2011-02-14 23:49:19 CET

1.0.21 release
----------------------------------------------------------------------------
r43588 | bednee | 2011-02-14 23:46:38 CET

Bugfixes: Missing reference to params cause {L=0} parameter reference not to work. When uriparameter was used in paramorder, it was appended as normal parameter too.
----------------------------------------------------------------------------
r41659 | bednee | 2010-12-31 16:19:43 CET

Multidomains issue fix, ,pre logging
----------------------------------------------------------------------------
r40463 | bednee | 2010-11-19 17:44:07 CET

Conflicts to other extensions added
----------------------------------------------------------------------------
r36134 | bednee | 2010-07-28 18:14:42 CEST

Major refactoring - methods splitted into smaller piecies for better readibility
----------------------------------------------------------------------------
r28316 | bednee | 2010-01-03 12:15:24 CET

Removing / from allowed characters, as it didn't pass rawurlencode untouched
----------------------------------------------------------------------------
r28315 | bednee | 2010-01-03 12:04:56 CET

Removing extra comma
----------------------------------------------------------------------------
r28314 | bednee | 2010-01-03 12:04:02 CET

Commiting missing SQL file for the Exclude always feature
----------------------------------------------------------------------------
r28313 | bednee | 2010-01-03 12:02:00 CET

Fixing regular expression
----------------------------------------------------------------------------
r28192 | bednee | 2009-12-29 21:45:49 CET

Special tt_news query added to FAQ section
----------------------------------------------------------------------------
r28191 | bednee | 2009-12-29 21:45:20 CET

Ereg_replace replaced by Preg_replace because of deprecation in PHP
----------------------------------------------------------------------------
r28190 | bednee | 2009-12-29 21:44:42 CET

Exclude always option added
----------------------------------------------------------------------------
r28189 | bednee | 2009-12-29 20:46:11 CET

Fixing doble slashes for GET vars
----------------------------------------------------------------------------
r25102 | bednee | 2009-10-05 21:19:48 CEST

Added another log
----------------------------------------------------------------------------
r24758 | bednee | 2009-09-23 22:34:45 CEST

Better logging
----------------------------------------------------------------------------
r24757 | bednee | 2009-09-23 22:26:01 CEST

Added some debug

----------------------------------------------------------------------------
r19885 | bednee | 2009-05-04 22:28:28 CEST

Fix when CoolUri is used in subdirectories
----------------------------------------------------------------------------
r19038 | bednee | 2009-04-15 20:11:51 CEST

Fix for empty $params
----------------------------------------------------------------------------
r18870 | bednee | 2009-04-09 00:12:42 CEST

"All" link now working
----------------------------------------------------------------------------
r18869 | bednee | 2009-04-09 00:12:19 CEST

Domain can be set from outside
----------------------------------------------------------------------------
r18868 | bednee | 2009-04-09 00:11:39 CEST

Version update
----------------------------------------------------------------------------
r18867 | bednee | 2009-04-09 00:11:22 CEST

1.0.16 RELEASE - manual updates
----------------------------------------------------------------------------
r17799 | bednee | 2009-03-14 00:21:05 CET


ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "cooluri".
 *
 * Auto generated 24-02-2014 23:52
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = array(
	'title' => 'CoolURI',
	'description' => 'RealURL alternative. Have nice URLs instead of ugly with parameters. CoolURI has user-friendly XML configuration file. For simple setup, just use the one supplied with extension and you are ready to go.',
	'category' => 'fe',
	'shy' => 0,
	'version' => '1.0.36',
	'dependencies' => '',
	'conflicts' => '',
	'priority' => '',
	'loadOrder' => '',
	'module' => '',
	'state' => 'stable',
	'uploadfolder' => 0,
	'createDirs' => '',
	'modify_tables' => '',
	'clearcacheonload' => 0,
	'lockType' => '',
	'author' => 'Jan Bednarik',
	'author_email' => 'info@bednarik.org',
	'author_company' => '',
	'CGLcompliance' => '',
	'CGLcompliance_note' => '',
	'constraints' => array(
		'depends' => array(
			'typo3' => '4.5.0-6.2.99',
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'_md5_values_when_last_written' => 'a:46:{s:9:"ChangeLog";s:4:"3b80";s:20:"class.tx_cooluri.php";s:4:"f16a";s:21:"ext_conf_template.txt";s:4:"8812";s:12:"ext_icon.gif";s:4:"4b29";s:17:"ext_localconf.php";s:4:"cdb9";s:17:"ext_tables-pg.sql";s:4:"08a4";s:14:"ext_tables.php";s:4:"4c55";s:14:"ext_tables.sql";s:4:"ecc2";s:25:"ext_tables_static+adt.sql";s:4:"508a";s:16:"locallang_db.xml";s:4:"929e";s:10:"README.txt";s:4:"cfff";s:31:"cooluri/CoolUriConf.xml_default";s:4:"364a";s:19:"cooluri/link.DB.php";s:4:"8121";s:24:"cooluri/link.DBLayer.php";s:4:"903e";s:26:"cooluri/link.Exception.php";s:4:"dc79";s:26:"cooluri/link.Functions.php";s:4:"d034";s:20:"cooluri/link.Log.php";s:4:"392b";s:21:"cooluri/link.Main.php";s:4:"9990";s:26:"cooluri/link.Translate.php";s:4:"26e8";s:20:"cooluri/link.URI.php";s:4:"f50b";s:21:"cooluri/XMLSchema.xml";s:4:"b0ea";s:36:"cooluri/manager/linkmanager.Main.php";s:4:"ab5e";s:40:"cooluri/manager/linkmanager.UserAuth.php";s:4:"cc97";s:14:"doc/manual.sxw";s:4:"30b0";s:19:"doc/wizard_form.dat";s:4:"6313";s:20:"doc/wizard_form.html";s:4:"fb6f";s:14:"mod1/clear.gif";s:4:"cc11";s:13:"mod1/conf.php";s:4:"0e2b";s:14:"mod1/index.php";s:4:"0cb7";s:18:"mod1/locallang.xml";s:4:"4a0d";s:22:"mod1/locallang_mod.xml";s:4:"2d90";s:22:"mod1/mod_template.html";s:4:"2328";s:19:"mod1/moduleicon.gif";s:4:"8bf5";s:14:"mod1/style.css";s:4:"f2a1";s:24:"mod1/img/button_edit.gif";s:4:"3248";s:27:"mod1/img/button_garbage.gif";s:4:"90c6";s:27:"mod1/img/button_refresh.gif";s:4:"5178";s:26:"mod1/img/button_sticky.gif";s:4:"b33f";s:13:"mod2/conf.php";s:4:"c268";s:14:"mod2/index.php";s:4:"12dc";s:22:"mod2/locallang_mod.xml";s:4:"99d2";s:22:"mod2/mod_template.html";s:4:"923b";s:19:"mod2/moduleicon.gif";s:4:"8bf5";s:33:"test/class.tx_cooluritest_pi1.php";s:4:"563a";s:20:"test/CoolUriConf.xml";s:4:"d86e";s:23:"test/test.Functions.php";s:4:"05f1";}',
	'suggests' => array(
	),
);

?>